src/middlewared/middlewared/alert/source/certificates.py

```diff
@@ -0,0 +1,26 @@
+from datetime import datetime
+
+from middlewared.alert.base import Alert, AlertLevel, AlertSource
+
+
+class CertRenewalAlertSource(AlertSource):
+    title = 'Certificate expiring'
+    level = AlertLevel.INFO
+
+    async def check(self):
+        alerts = []
+
+        for cert in await self.middleware.call(
+            'certificate.query',
+            [['certificate', '!=', None]]
+        ) + await self.middleware.call('certificateauthority.query'):
+            diff = (datetime.strptime(cert['until'], '%a %b %d %H:%M:%S %Y') - datetime.now()).days
+            if diff < 10:
+                alerts.append(
+                    Alert(
+                        title=f'{cert["name"]} expiring within {diff} days',
```

Comment:
`Certificate #1 expires in 6 days` and `Certificate #1 expires in 5 days` would be considered different alerts. Each time the day changes user would receive a notification that alert `Certificate #1 expires in 6 days` is gone and there is new alert `Certificate #1 expires in 5 days`. We don't want that.

Instead, we do
```
Alert(
    title=f'{cert["name"]} expiring within {diff} days',
    key=[cert["name"]],
)
```

This way system will now that the only valuable part in the alert description is `cert["name"]` and won't bother the user.

Even better idea is to include alert severity in the key:

```
Alert(
    title=f'{cert["name"]} expiring within {diff} days',
    key=[cert["name"], diff < 2],
)
```

So if user dismisses this alert and forgets about it, he will be reminded second time when alert becomes critical.